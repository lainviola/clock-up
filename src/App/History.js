// @flow
import React, { Component } from 'react'
import styled from 'styled-components'
import { borderColor, textColor } from '../common/CSS'
import { calcTotalTime, ISOtoYmd, ISOtoHm } from '../common/util'
import { Table, Tr, Td, Tbody, Th } from '../common/components/Table'
import { Select } from '../common/components/Select'

type Props = {
  clocks: Array<Object>
}

export class History extends Component<Props> {
  render() {
    const { clocks } = this.props

    var history = []
    if (clocks.length) {
      history = clocks
        // must have complate clock in/out. not allow only clock in data of first time.
        .filter(v => {
          return v.clockOut !== null // during clock-in data. can not calculate totalTime when clockout was null.
        })
        .map((v, i) => {
          const clockIn = v.clockIn
          const clockout = v.clockOut
          const createdAt = v.createdAt
          const small = new Date(clockIn)
          const large = new Date(clockout)
          const total = calcTotalTime(large, small)

          return (
            <tr key={i}>
              <Td>{ISOtoYmd(createdAt)}</Td>
              <Td>{total}</Td>
              <Td>{ISOtoHm(clockIn)}</Td>
              <Td>{ISOtoHm(clockout)}</Td>
            </tr>
          )
        })
    }

    // clocks.length == 0 or clocks.lengh ===1 and clocks.[0] only has clockIn value without clockOut.
    if (history.length === 0) {
      history = (
        <Tr>
          <Td>N/A</Td>
          <Td>N/A</Td>
          <Td>N/A</Td>
          <Td>N/A</Td>
        </Tr>
      )
    }

    // TODO 表示期間を変更するセレクトボックスを追加
    return (
      <Container>
        <Header>History</Header>
        <SelectBoxWrapper>
          <Select>
            <option>latest 1week</option>
            <option selected>latest 1month</option>
            <option>2018/01</option>
          </Select>
        </SelectBoxWrapper>
        <Table>
          <Tbody>
            <Tr>
              <Th>date</Th>
              <Th>total</Th>
              <Th>in</Th>
              <Th>out</Th>
            </Tr>
            {history}
          </Tbody>
        </Table>
      </Container>
    )
  }
}

const Container = styled.div`
  color: ${textColor};
`

const SelectBoxWrapper = styled.div`
  padding: 10px;
  display: flex;
  flex-direction: row;
  justify-content: center;
`

const Header = styled.div`
  padding: 10px 0;
  text-align: center;
  font-size: 1.2em;
  border-bottom: 1px solid ${borderColor};
`

export default History
