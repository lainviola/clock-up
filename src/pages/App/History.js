// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import styled from 'styled-components'
import gql from 'graphql-tag'
import { graphql } from 'react-apollo'
import { compose } from 'redux'
import { borderColor, textColor } from '../../css'
import { calcTotalTime, ISOtoYmd, ISOtoHm } from '../../util'
import { Table, Tr, Td, Tbody, Th } from '../../elements/Table'
import { Select } from '../../elements/Select'
import type { HistoryQueryParameter } from '../../types/HistoryQueryParameter'
import type { ReduxState } from '../../types/ReduxState'

const Container = styled.div`
  color: ${textColor};
`

const SelectBoxWrapper = styled.div`
  padding: 10px;
  display: flex;
  flex-direction: row;
  justify-content: center;
`

const Header = styled.div`
  padding: 10px 0;
  text-align: center;
  font-size: 1.2em;
  border-bottom: 1px solid ${borderColor};
`

type Clocks = Array<{
  id: string,
  clockIn: string,
  clockOut: string,
  createdAt: string,
  updatedAt: string
}>

type User = {
  clocks: Clocks
}

type GraqhQLData = {
  user: User,
  loading: boolean
}

type Props = {
  data: GraqhQLData,
  historyQueryParameter: HistoryQueryParameter
}

export class History extends Component<Props> {
  // TODO to be implement
  renewGQL() {
    console.log('renewGQL')
  }

  render() {
    const { loading } = this.props.data
    if (loading) return null

    const { clocks } = this.props.data.user

    var history = []
    if (clocks.length) {
      history = clocks
        // must have complate clock in/out. not allow only clock in data of first time.
        .filter(v => {
          return v.clockOut !== null // during clock-in data. can not calculate totalTime when clockout was null.
        })
        .map((v, i) => {
          const clockIn = v.clockIn
          const clockout = v.clockOut
          const createdAt = v.createdAt
          const small = new Date(clockIn)
          const large = new Date(clockout)
          const total = calcTotalTime(large, small)

          return (
            <tr key={i}>
              <Td>{ISOtoYmd(createdAt)}</Td>
              <Td>{total}</Td>
              <Td>{ISOtoHm(clockIn)}</Td>
              <Td>{ISOtoHm(clockout)}</Td>
            </tr>
          )
        })
    }

    // clocks.length == 0 or clocks.lengh ===1 and clocks.[0] only has clockIn value without clockOut.
    if (history.length === 0) {
      history = (
        <Tr>
          <Td>N/A</Td>
          <Td>N/A</Td>
          <Td>N/A</Td>
          <Td>N/A</Td>
        </Tr>
      )
    }

    return (
      <Container>
        <Header>History</Header>
        <SelectBoxWrapper>
          <Select onChange={this.renewGQL} defaultValue={'1week'}>
            <option value="1week">1week</option>
            <option value="1month">1month</option>
            <option>2018/01</option>
          </Select>
        </SelectBoxWrapper>
        <Table>
          <Tbody>
            <Tr>
              <Th>date</Th>
              <Th>total</Th>
              <Th>in</Th>
              <Th>out</Th>
            </Tr>
            {history}
          </Tbody>
        </Table>
      </Container>
    )
  }
}

const UserClocksQuery = gql`
  query FetchUserClocks($first: Int, $orderBy: ClockOrderBy) {
    user {
      id
      clocks(first: $first, orderBy: $orderBy) {
        id
        clockIn
        clockOut
        createdAt
        updatedAt
      }
    }
  }
`

const mapStateToProps = (state: ReduxState) => {
  return { historyQueryParameter: state.app.historyQueryParameter }
}

export default compose(
  connect(mapStateToProps),
  graphql(UserClocksQuery, {
    options: ({ historyQueryParameter }) => {
      return {
        variables: {
          first: historyQueryParameter.first,
          orderBy: historyQueryParameter.orderBy
        }
      }
    }
  })
)(History)
