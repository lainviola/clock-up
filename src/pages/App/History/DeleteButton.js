// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { graphql, withApollo } from 'react-apollo'
import { pure, compose } from 'recompose'
import styled from 'styled-components'
import { Button } from '../../../components/Button'
import { theme } from '../../../theme'
import { DELETE_CLOCK_MUTATION } from '../../../graphql/mutation'
import type { MutationFunc } from 'react-apollo'
import { HISTORY_BOARD_QUERY } from '../../../graphql/query'
import type { Dispatch } from 'redux'
import type { ON_CLICK_HISTORY_DELETE_BUTTON } from '../../../types/action'

const Container = styled.div`
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`

type Props = {
  dispatch: Dispatch<ON_CLICK_HISTORY_DELETE_BUTTON>,
  checkedHistoryIdList: Array<string>,
  DELETE_CLOCK_MUTATION: MutationFunc<*, *>
}

class DeleteButton extends Component<Props> {
  handleClick = (): void => {
    const { checkedHistoryIdList, DELETE_CLOCK_MUTATION, dispatch } = this.props
    dispatch({ type: 'ON_CLICK_HISTORY_DELETE_BUTTON' })

    checkedHistoryIdList.forEach(id => {
      DELETE_CLOCK_MUTATION({
        variables: { clockId: id },

        /**
         * START: refrect mutation to UI
         */
        update: (cache, { data: { deleteClock } }) => {
          const data = cache.readQuery({
            query: HISTORY_BOARD_QUERY
          })
          // $FlowIssue
          const oldClicks = data.user.clocks
          const newClocks = oldClicks.filter(v => v.id !== id)

          // hack https://stackoverflow.com/questions/27519836/uncaught-typeerror-cannot-assign-to-read-only-property
          // $FlowIssue
          const newUser = { ...data.user, writable: true }
          const newData = { user: newUser }
          newData.user.clocks = newClocks
          // hack https://stackoverflow.com/questions/27519836/uncaught-typeerror-cannot-assign-to-read-only-property

          cache.writeQuery({
            query: HISTORY_BOARD_QUERY,
            data: newData
          })
          /**
           * END: refrect mutation to UI
           */
        }
      })
    })
  }

  render() {
    const { checkedHistoryIdList } = this.props

    return (
      <Container>
        {checkedHistoryIdList.length ? (
          <Button onClick={this.handleClick} color={theme.red}>
            Delete
          </Button>
        ) : null}
      </Container>
    )
  }
}

export default compose(
  connect(),
  withApollo,
  graphql(DELETE_CLOCK_MUTATION, { name: 'DELETE_CLOCK_MUTATION' }),
  pure
)(DeleteButton)
