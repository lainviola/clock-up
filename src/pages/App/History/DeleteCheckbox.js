// @flow
import React, { Component } from 'react'
import { pure, compose } from 'recompose'
import { connect } from 'react-redux'
import styled, { css } from 'styled-components'
import { theme } from '../../../theme'
import type { Dispatch } from 'redux'
import type {
  CHECK_DELETE_HISTORY,
  UNCHECK_DELETE_HISTORY
} from '../../../types/action'
import type { ReduxState } from '../../../reducer'

const Checkbox = styled.div`
  width: 16px;
  height: 16px;
  border-radius: 100%;
  border-style: solid;
  border-width: 1px;
  border-color: ${props => (props.checked ? theme.red : 'rgba(0, 0, 0, 0.2)')};
  ${props =>
    props.checked &&
    css`
      background-color: ${theme.red};
    `}
  }
`

type StateProps = {|
  checkedHistoryIdList: Array<string>
|}

type Props = {
  ...StateProps,
  clockId: string,
  dispatch: Dispatch<CHECK_DELETE_HISTORY | UNCHECK_DELETE_HISTORY>
}

class DeleteCheckbox extends Component<Props> {
  isChecked = (clockId: string): boolean => {
    const { checkedHistoryIdList } = this.props

    return checkedHistoryIdList.includes(clockId)
  }

  handleClick = () => {
    const { dispatch, clockId } = this.props

    if (this.isChecked(clockId)) {
      dispatch({
        type: 'UNCHECK_DELETE_HISTORY',
        clockId: clockId
      })
    } else {
      dispatch({
        type: 'CHECK_DELETE_HISTORY',
        clockId: clockId
      })
    }
  }

  render() {
    return (
      <Checkbox
        onClick={this.handleClick}
        checked={this.isChecked(this.props.clockId)}
      />
    )
  }
}

const map: MapStateToProps = (state: ReduxState): StateProps => {
  return { checkedHistoryIdList: state.checkedHistoryIdList }
}

export default compose(
  connect(map),
  pure
)(DeleteCheckbox)
