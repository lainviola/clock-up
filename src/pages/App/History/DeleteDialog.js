// @flow
import React, { Component } from 'react'
import styled from 'styled-components'
import { compose, pure } from 'recompose'
import { HISTORY_BOARD_QUERY } from '../../../graphql/query'
import { connect } from 'react-redux'
import { graphql, withApollo } from 'react-apollo'
import type { CLICK_HISTORY_DELETE_BUTTON } from '../../../types/action'
import { DELETE_CLOCK_MUTATION } from '../../../graphql/mutation'
import type { MutationFunc } from 'react-apollo'
import type { Dispatch } from 'redux'

const Container = styled.div``

type Props = {
  dispatch: Dispatch<CLICK_HISTORY_DELETE_BUTTON>,
  checkedHistoryIdList: Array<string>,
  DELETE_CLOCK_MUTATION: MutationFunc<*, *>
}

class DeleteDialog extends Component<Props> {
  handleClick = (): void => {
    const { checkedHistoryIdList, DELETE_CLOCK_MUTATION, dispatch } = this.props
    dispatch({ type: 'CLICK_HISTORY_DELETE_BUTTON' })

    checkedHistoryIdList.forEach(id => {
      DELETE_CLOCK_MUTATION({
        variables: { clockId: id },

        /**
         * START: refrect mutation to UI
         */
        update: (cache, { data: { deleteClock } }) => {
          const data = cache.readQuery({
            query: HISTORY_BOARD_QUERY
          })
          // $FlowIssue
          const oldClicks = data.user.clocks
          const newClocks = oldClicks.filter(v => v.id !== id)

          // hack https://stackoverflow.com/questions/27519836/uncaught-typeerror-cannot-assign-to-read-only-property
          // $FlowIssue
          const newUser = { ...data.user, writable: true }
          const newData = { user: newUser }
          newData.user.clocks = newClocks
          // hack https://stackoverflow.com/questions/27519836/uncaught-typeerror-cannot-assign-to-read-only-property

          cache.writeQuery({
            query: HISTORY_BOARD_QUERY,
            data: newData
          })
          /**
           * END: refrect mutation to UI
           */
        }
      })
    })
  }

  render() {
    return (
      <Container>
        <button onclick={this.handleClick()}>delete</button>
      </Container>
    )
  }
}

export default compose(
  connect(),
  c => withApollo<*>(c),
  graphql(DELETE_CLOCK_MUTATION, { name: 'DELETE_CLOCK_MUTATION' }),
  pure
)(DeleteDialog)
