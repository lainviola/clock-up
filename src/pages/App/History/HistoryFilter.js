// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import { pure, compose } from 'recompose'
import styled from 'styled-components'
import type { Period } from '../../../dataTypes'
import type { Dispatch } from 'redux'
import type { ON_CHANGE_HISTORY_FILTER } from '../../../actionTypes'

const Container = styled.div`
  flex: 1;
  display: flex;
  justify-content: center;
  align-items: center;
`

// TODO style
const StyledSelect = styled.select``

type Props = {
  dispatch: Dispatch<ON_CHANGE_HISTORY_FILTER>
}

class HistoryFilter extends Component<Props> {
  renewGQL(value: Period) {
    this.props.dispatch({
      type: 'ON_CHANGE_HISTORY_FILTER',
      period: value
    })
  }

  render() {
    return (
      <Container>
        <StyledSelect
          onChange={e => this.renewGQL(e.target.value)}
          defaultValue={'1week'}
        >
          <option value="1week">1week</option>
          <option value="1month">1month</option>
          <option value="all">all</option>
        </StyledSelect>
      </Container>
    )
  }
}

export default compose(
  connect(),
  pure
)(HistoryFilter)
