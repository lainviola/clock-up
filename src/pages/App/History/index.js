// @flow
import React, { Component } from 'react'
import { connect } from 'react-redux'
import gql from 'graphql-tag'
import { graphql } from 'react-apollo'
import { compose } from 'redux'
import { calcTotalTime, ISOtoYmd, ISOtoHm } from '../../../function'
import { Table, Tr, Td, Tbody, Th } from '../../../elements/Table'
import { Select } from '../../../elements/Select'
import InTime from './InTime'
import OutTime from './OutTime'
import { Container, SelectBoxWrapper, Header } from './index.style'
import type { Dispatch } from 'redux'
import type { Period } from '../../../dataType'
import type { ReduxAction } from '../../../action'
import type { HistoryQueryParameter } from '../../../dataType'
import type { ReduxState } from '../../../reducer'

type Clocks = Array<{
  id: string,
  clockIn: string,
  clockOut: string,
  createdAt: string,
  updatedAt: string
}>

type StateProps = {|
  historyQueryParameter: HistoryQueryParameter
|}

type Props = StateProps & {
  data: {
    user: { clocks: Clocks },
    loading: boolean
  },
  dispatch: Dispatch<ReduxAction>
}

export class History extends Component<Props> {
  renewGQL(value: Period) {
    this.props.dispatch({
      type: 'CHANGE_HISTORY',
      period: value
    })
  }

  render() {
    const { loading } = this.props.data
    if (loading) return null

    const { clocks } = this.props.data.user

    var history = []
    if (clocks.length) {
      history = clocks
        // must have complate clock in/out. not allow only clock in data of first time.
        .filter(v => {
          return v.clockOut !== null // during clock-in data. can not calculate totalTime when clockout was null.
        })
        .map((v, i) => {
          const clockIn = v.clockIn
          const clockout = v.clockOut
          const createdAt = v.createdAt
          const small = new Date(clockIn)
          const large = new Date(clockout)
          const total = calcTotalTime(large, small)

          return (
            <tr key={i} enzyme-testid={`history-table-time-${i}`}>
              <Td>{ISOtoYmd(createdAt)}</Td>
              <Td>{total}</Td>
              <InTime date={ISOtoHm(clockIn)} />
              <OutTime date={ISOtoHm(clockout)} />
            </tr>
          )
        })
    }

    // clocks.length == 0 or clocks.lengh ===1 and clocks.[0] only has clockIn value without clockOut.
    if (history.length === 0) {
      history = (
        <Tr enzyme-testid="history-table-na">
          <Td>N/A</Td>
          <Td>N/A</Td>
          <Td>N/A</Td>
          <Td>N/A</Td>
        </Tr>
      )
    }

    return (
      <Container>
        <Header>History</Header>
        <SelectBoxWrapper>
          <Select
            onChange={e => this.renewGQL(e.target.value)}
            defaultValue={'1week'}
          >
            <option value="1week">1week</option>
            <option value="1month">1month</option>
            <option value="all">all</option>
          </Select>
        </SelectBoxWrapper>
        <Table enzyme-testid="history-table">
          <Tbody>
            <Tr>
              <Th>date</Th>
              <Th>total</Th>
              <Th>in</Th>
              <Th>out</Th>
            </Tr>
            {history}
          </Tbody>
        </Table>
      </Container>
    )
  }
}

const UserClocksQuery = gql`
  query FetchUserClocks($first: Int, $orderBy: ClockOrderBy) {
    user {
      id
      clocks(first: $first, orderBy: $orderBy) {
        id
        clockIn
        clockOut
        createdAt
        updatedAt
      }
    }
  }
`

const mapStateProps = (state: ReduxState): StateProps => {
  return { historyQueryParameter: state.historyQueryParameter }
}

export default compose(
  connect(mapStateProps),
  graphql(UserClocksQuery, {
    options: ({ historyQueryParameter }) => {
      // $FlowFixMe
      return {
        variables: {
          first: historyQueryParameter.first,
          orderBy: historyQueryParameter.orderBy
        }
      }
    }
  })
)(History)
