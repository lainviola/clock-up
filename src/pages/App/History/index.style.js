import styled from 'styled-components'
import { theme } from '../../../theme'
import { Td, Th } from '../../../components/Table'

export const Container = styled.div`
  color: ${theme.textColor};
  padding-left: 5px;
  padding-right: 5px;
`

export const ControlArea = styled.div`
  padding: 10px;
  height: 40px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`

export const Header = styled.div`
  padding: 10px 0;
  text-align: center;
  font-size: 1.2em;
  border-bottom: 1px solid ${theme.borderColor};
`

export const DeleteCheckboxTh = styled(Th)`
  width: 30px;
  border-width: 0;
  border-top-width: 1px;
  border-top-color: #fff;
  border-top-style: solid;
`

export const DeleteCheckboxTd = styled(Td)`
  width: 30px;
  border-width: 0;
  background-color: #fff !important;
  display: flex;
  justify-content: center;
  align-items: center;
`
