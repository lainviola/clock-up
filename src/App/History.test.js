import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { History } from './History'

describe('<History />', () => {
  describe('pass empty array', () => {
    function setup() {
      const wrapper = shallow(<History clocks={[]} />)
      return wrapper
    }

    it('should be render without error', () => {
      const wrapper = setup()

      expect(wrapper.exists()).toBe(true)
    })

    it('should be render with N/A', () => {
      const wrapper = setup()

      expect(wrapper.find('Table__Td').exists()).toBe(true)
      expect(
        wrapper
          .find('Table__Td')
          .first()
          .dive()
          .exists()
      ).toBe(true)
      expect(
        wrapper
          .find('Table__Td')
          .first()
          .dive()
          .text()
      ).toBe('N/A')
    })

    it('should be match snapshot', () => {
      const wrapper = setup()

      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
  describe('pass first time clock in data', () => {
    const clocks = [
      {
        id: 'cjbfuqo1j3n5p0146x0pjsj7o',
        clockIn: '2017-12-21T02:12:28.707Z',
        clockOut: null,
        createdAt: '2017-12-21T02:12:28.000Z',
        updatedAt: '2017-12-21T02:12:28.000Z',
        __typename: 'Clock'
      }
    ]
    function setup() {
      const wrapper = shallow(<History clocks={clocks} />)
      return wrapper
    }

    it('should be render with N/A', () => {
      const wrapper = setup()

      expect(wrapper.find('Table__Td').exists()).toBe(true)
      expect(
        wrapper
          .find('Table__Td')
          .first()
          .dive()
          .exists()
      ).toBe(true)
      expect(
        wrapper
          .find('Table__Td')
          .first()
          .dive()
          .text()
      ).toBe('N/A')
    })

    it('should be match snapshot', () => {
      const wrapper = setup()

      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
  describe('pass complete data', () => {
    const clocks = [
      {
        id: 'cjbhtpf8psjmv0146zkmd9zrh',
        clockIn: '2017-12-22T11:19:02.776Z',
        clockOut: null,
        createdAt: '2017-12-22T11:19:03.000Z',
        updatedAt: '2017-12-22T11:19:03.000Z',
        __typename: 'Clock'
      },
      {
        id: 'cjbfuqo1j3n5p0146x0pjsj7o',
        clockIn: '2017-12-21T02:12:28.707Z',
        clockOut: '2017-12-21T16:54:21.094Z',
        createdAt: '2017-12-21T02:12:28.000Z',
        updatedAt: '2017-12-21T16:54:21.000Z',
        __typename: 'Clock'
      }
    ]
    function setup() {
      const wrapper = shallow(<History clocks={clocks} />)
      return wrapper
    }

    it('should be render TotalTime', () => {
      const wrapper = setup()

      expect(wrapper.find('tr').exists()).toBe(true)
      expect(wrapper.html().includes('2017/12/21')).toBe(true) // "date" column
      expect(wrapper.html().includes('14h41m')).toBe(true) // "total" column
      expect(wrapper.html().includes('11:12')).toBe(true) // "in" column
      expect(wrapper.html().includes('01:54')).toBe(true) // "out" column
    })

    it('should be match snapshot', () => {
      const wrapper = setup()

      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })
})
