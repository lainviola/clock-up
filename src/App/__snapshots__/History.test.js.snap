// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> pass empty array should be match snapshot 1`] = `
<History__Container>
  <History__Header>
    History
  </History__Header>
  <History__SelectBoxWrapper>
    <Select>
      <option>
        latest 1week
      </option>
      <option
        selected={true}
      >
        latest 1month
      </option>
      <option>
        2018/01
      </option>
    </Select>
  </History__SelectBoxWrapper>
  <Table>
    <Table__Tbody>
      <Table__Tr>
        <Table__Th>
          date
        </Table__Th>
        <Table__Th>
          total
        </Table__Th>
        <Table__Th>
          in
        </Table__Th>
        <Table__Th>
          out
        </Table__Th>
      </Table__Tr>
      <Table__Tr>
        <Table__Td>
          N/A
        </Table__Td>
        <Table__Td>
          N/A
        </Table__Td>
        <Table__Td>
          N/A
        </Table__Td>
        <Table__Td>
          N/A
        </Table__Td>
      </Table__Tr>
    </Table__Tbody>
  </Table>
</History__Container>
`;

exports[`<History /> pass first time clock in data should be match snapshot 1`] = `
<History__Container>
  <History__Header>
    History
  </History__Header>
  <History__SelectBoxWrapper>
    <Select>
      <option>
        latest 1week
      </option>
      <option
        selected={true}
      >
        latest 1month
      </option>
      <option>
        2018/01
      </option>
    </Select>
  </History__SelectBoxWrapper>
  <Table>
    <Table__Tbody>
      <Table__Tr>
        <Table__Th>
          date
        </Table__Th>
        <Table__Th>
          total
        </Table__Th>
        <Table__Th>
          in
        </Table__Th>
        <Table__Th>
          out
        </Table__Th>
      </Table__Tr>
      <Table__Tr>
        <Table__Td>
          N/A
        </Table__Td>
        <Table__Td>
          N/A
        </Table__Td>
        <Table__Td>
          N/A
        </Table__Td>
        <Table__Td>
          N/A
        </Table__Td>
      </Table__Tr>
    </Table__Tbody>
  </Table>
</History__Container>
`;
